libs/storage/Tsavorite/cs/src/core/Index/Interfaces/TsavoriteEqualityComparer.cs

```diff
@@ -11,24 +11,23 @@ internal static class TsavoriteEqualityComparer
     {
         public static ITsavoriteEqualityComparer<T> Get<T>()
         {
-            var t = typeof(T);
-            if (t == typeof(string))
-                return new StringTsavoriteEqualityComparer() as ITsavoriteEqualityComparer<T>;
-            else if (t == typeof(byte[]))
-                return new ByteArrayTsavoriteEqualityComparer() as ITsavoriteEqualityComparer<T>;
-            else if (t == typeof(long))
-                return new LongTsavoriteEqualityComparer() as ITsavoriteEqualityComparer<T>;
-            else if (t == typeof(int))
-                return new IntTsavoriteEqualityComparer() as ITsavoriteEqualityComparer<T>;
-            else if (t == typeof(Guid))
-                return new GuidTsavoriteEqualityComparer() as ITsavoriteEqualityComparer<T>;
-            else if (t == typeof(SpanByte))
-                return new SpanByteComparer() as ITsavoriteEqualityComparer<T>;
+            if (typeof(T) == typeof(string))
+                return (ITsavoriteEqualityComparer<T>)(object)StringTsavoriteEqualityComparer.Instance;
```

Comment:
You might want to look at Unsafe.As here, instead of the (object) cast inbetween.

Reply:
This is a common pattern for generic bridging that JIT recognizes. Both would work.

https://grep.app/search?q=%28T%29%28object%29